### PR TITLE
Add create profile page functionality

### PR DESCRIPTION
**Pull request by brownm26csp**
Styled Create Profile page

Added functionality to automatically route user to "create profile" page if their username is not in the database (using a hardcoded list for now)

Added functionality to keep user on home page if their username is in the database

In this image, there is a "create profile button" which can be used to test a database connection, but which will not be used for the final product
image

In this image, there is a field where you can put a username. This is just for testing - but I have it set that if the user arrived to this page via clicking on "create an account," then the e-mail they entered using Auth0 will override anything they enter here. Again, this is just for testing - we won't have the "e-mail" field available for the final product
image

Pre-filled and disabled if user is signing up for first time:
image

### CODE REVIEW

File: Frontend/src/app/create-profile/create-profile.component.html [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+        </mat-form-field>
+        <mat-form-field color="warn">
+            <mat-label for="lastName">Last Name</mat-label>
+            <input matInput formControlName="lastName">
+        </mat-form-field>
+        <mat-form-field color="warn">
+            <mat-label for="bio">Bio</mat-label>
+            <input matInput formControlName="bio">
+        </mat-form-field>
+        <mat-form-field color="warn">
```
Comment (averytribbett) [inline Frontend/src/app/create-profile/create-profile.component.html:36]: I am not familiar with these tags. Is this a component library?

File: Frontend/src/app/create-profile/create-profile.component.html [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+            <mat-label for="username">E-mail (TEST ONLY)</mat-label>
+            <input matInput formControlName="username">
```
Comment (averytribbett) [inline Frontend/src/app/create-profile/create-profile.component.html]: One small optimization here is we could autofill the email if they have been authed with auth0 and disable the input. I think this is the standard UX approach so the user knows their email is already added.

File: Frontend/src/app/create-profile/create-profile.component.html (hunk 1 above)
Comment (brownm26csp) [reply Frontend/src/app/create-profile/create-profile.component.html:36]: Yeah, essentially the "mat" stuff is part of the angular materials library. So, it's basically like a "pre-formatted" input tag. I found it from here: https://material.angular.io/components/input/overview

Comment (averytribbett) [reply Frontend/src/app/create-profile/create-profile.component.html:36]: Ok awesome!

**Review by averytribbett — APPROVED** [review]

**Pull request merged**